deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/NeuralNetConfiguration.java

```diff
@@ -1008,17 +1008,23 @@ public Builder l2Bias(double l2Bias) {
         }
 
         /**
-         * Dropout probability. This is the probability of <it>retaining</it> an activation. So dropOut(x) will keep an
-         * activation with probability x, and set to 0 with probability 1-x.<br>
-         * dropOut(0.0) is disabled (default).
+         * Dropout probability. This is the probability of <it>retaining</it> an input activation for a layer. So
+         * dropOut(x) will keep an input activation with probability x, and set to 0 with probability 1-x.<br>
+         * dropOut(0.0) is disabled (default). When {@link #useDropConnect(boolean)} is set to true (false by default),
+         * this method sets the drop connect probability instead.
          * <p>
-         * Note: This sets the probability per-layer. Care should be taken when setting lower values for complex networks.
+         * Note 1: This sets the probability per-layer. Care should be taken when setting lower values for
+         * complex networks (too much information may be lost with aggressive dropout values).<br>
+         * Note 2: Frequently, dropout is not applied to input (first layer) our output layer. This needs to be
+         * handled MANUALLY by the user - set .dropout(0) on those layers when using global dropout setting.<br>
```

Comment:
Regarding input layer. The "dropout paper" (_the original!_) suggests 0.8 retain on input, 0.5 on hidden layers. So yes, it should definitely be different, but not necessarily "retain all".

AGAIN the point about "0": I consistently get a "huh?!"-feeling seeing that number - which implies that it should always be pointed out something like _".. thus disabling the dropout for this layer"_, or something.

Yes, I realize that I really don't like that special value, as it so directly contradicts the semantic of what the number means.